include/merlin_hashtable.cuh

```diff
@@ -295,25 +300,34 @@ class HashTable {
         load_factor = fast_load_factor();
       }
 
-      Selector::execute_kernel(
-          load_factor, options_.block_size, stream, n, c_table_index_, d_table_,
-          keys, reinterpret_cast<const value_type*>(values), metas);
+      Selector::execute_kernel(load_factor, options_.block_size, stream, n,
+                               c_table_index_, d_table_, keys, values, metas);
     } else {
-      const size_type dev_ws_size{n * (sizeof(value_type*) + sizeof(int))};
+      const size_type dev_ws_base_size{n * (sizeof(value_type*) + sizeof(int))};
+      const size_type dev_ws_size{dev_ws_base_size +
+                                  (ext_store_ ? n : 0) * sizeof(key_type)};
       auto dev_ws{dev_mem_pool_->get_workspace<1>(dev_ws_size, stream)};
       auto d_dst{dev_ws.get<value_type**>(0)};
       auto d_src_offset{reinterpret_cast<int*>(d_dst + n)};
+      auto d_evicted_keys{reinterpret_cast<key_type*>(d_src_offset + n)};
 
-      CUDA_CHECK(cudaMemsetAsync(d_dst, 0, dev_ws_size, stream));
+      CUDA_CHECK(cudaMemsetAsync(d_dst, 0, dev_ws_base_size, stream));
```

Comment:
Is this change the part of GDS API design? If not, please move them to a separate PR.

Reply:
Done